### PR TITLE
Lionelb/indexing crawler bot

### PR DESCRIPTION
**Pull request by lionelB**
This PR allow to handle crawling bots depending on the ENVIRONMENT CI flag.
- [x] block all content from being indexed by bot on dev / staging deployement
- [x] instruct bot to noindex but follow search results pages using meta
- [x] add a canonical meta on content coming from other sources

### CODE REVIEW

File: packages/code-du-travail-frontend/server.js [hunk 1]
```diff
@@ -3,18 +3,50 @@ const express = require("express");
 const next = require("next");
 const routes = require("./routes");
 
+/**
+ * this env variable is use to target developpement / staging deployement
+ * in order to block indexing bot using a x-robot-header and an appropriate robots.txt
+ */
+const LIVEPROD = process.env.ENVIRONMENT === "production";
```
Comment (UnbearableBear) [inline packages/code-du-travail-frontend/server.js:10]: Don't we use the NODE_ENV variable ? I thought it was some sort of convention

Comment (lionelB) [reply packages/code-du-travail-frontend/server.js:10]: NODE_ENV is generally used to build a "production" bundle
we use production bundle for  dev / staging environment
What I need is to know if the env is the production site (aka codedutravail.num.social.gouv.fr) or any other deployement (aka codedutravail-dev.num.social.gouv.fr, xxxxxx.code-du-travail-numerique.dev.factory.social.gouv.fr) to handle the robots.txt accordingly

File: packages/code-du-travail-frontend/pages/fiche-service-public.js [hunk 2]
```diff
@@ -66,6 +67,9 @@ class Fiche extends React.Component {
           description={description}
           image={ogImage}
         />
+        <Head>
+          <link href={data._source.url} rel="canonical" />
```
Comment (UnbearableBear) [inline packages/code-du-travail-frontend/pages/fiche-service-public.js:71]: Since you disallow robots on dev env do we really need the canonical tag ?

**Review by UnbearableBear — APPROVED** [review]

File: packages/code-du-travail-frontend/pages/fiche-service-public.js (hunk 2 above)
Comment (lionelB) [reply packages/code-du-travail-frontend/pages/fiche-service-public.js:71]: the canonical stuff is used to instruct search engine that the content here come from another site.

Comment (UnbearableBear) [reply packages/code-du-travail-frontend/pages/fiche-service-public.js:71]: So you are telling google that it should index pages on the other domains and not ours ? That's very opinionated in my opinion ! Is that really what we want ?

Comment (lionelB) [reply packages/code-du-travail-frontend/pages/fiche-service-public.js:71]: > So you are telling google that it should index pages on the other domains and not ours ? That's very opinionated in my opinion ! Is that really what we want ?

I don't intend to says that. Just that the content is coming from another site

**Pull request merged**